packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts

```diff
@@ -180,13 +180,14 @@ export class FauxClassGenerator {
   }
 
   // VisibleForTesting
-  public generateFauxClass(folderPath: string, sobject: any): string {
+  public generateFauxClass(folderPath: string, sobject: SObject): string {
     if (!fs.existsSync(folderPath)) {
       fs.mkdirSync(folderPath);
     }
     const fauxClassPath = path.join(folderPath, sobject.name + '.cls');
-
-    fs.writeFileSync(fauxClassPath, this.generateFauxClassText(sobject));
+    if (sobject.name) {
+      fs.writeFileSync(fauxClassPath, this.generateFauxClassText(sobject));
+    }
 
     return fauxClassPath;
```

Comment:
Why is this returning the "undefined.cls" then? What is the reasoning for that? Is that a potential bug?

Reply:
I just deleted my org so I can't get the exact error anymore, but somehow some of the sobjects from sobject list return an error with "object not found" or something similar.

Reply:
The generation assumes it's a valid sobject, so only line 187, it will just give the path to be ../undefined.cls when sobject.name is undefined.

Reply:
Actually I'll move this check to a better place so it won't even come into this method if it's an invalid sobject